### PR TITLE
nixUnstable: pre20210802_47e96bb -> pre20210830_50a3586

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

Deliver recent fixes.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by roberth** [comment]
Working on the darwin failure...

**Comment by happysalada** [comment]
I'm getting the following error trying to build on darwin
```
nix> build flags: -j4 -l4 SHELL=/nix/store/q0vakbapiqc42icmhzx7iy54smj9293n-bash-4.4-p23/bin/bash profiledir=\$\(out\)/etc/profile.d
nix>   GEN    Makefile.config
nix>   GEN    src/libexpr/parser-tab.cc
nix>   GEN    src/libexpr/lexer-tab.hh
nix>   GEN    src/nix-channel/unpack-channel.nix.gen.hh
nix>   GEN    src/nix-env/buildenv.nix.gen.hh
nix>   GEN    src/nix/get-env.sh.gen.hh
nix>   GEN    src/libexpr/flake/call-flake.nix.gen.hh
nix>   GEN    src/libexpr/imported-drv-to-derivation.nix.gen.hh
nix>   GEN    src/libexpr/primops/derivation.nix.gen.hh
nix>   GEN    src/libexpr/fetchurl.nix.gen.hh
nix>   CXX    src/libutil/affinity.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/archive.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/args.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/compression.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/compute-levels.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/config.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/error.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/hash.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/json.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/logging.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/rust-ffi.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/serialise.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix> src/libutil/serialise.cc:247:14: warning: 'finish' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
nix>         void finish() {
nix>              ^
nix> src/libutil/serialise.hh:31:18: note: overridden virtual function is here
nix>     virtual void finish() = 0;
nix>                  ^
nix> 1 warning generated.
nix>   CXX    src/libutil/tarfile.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/thread-pool.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/url.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix>   CXX    src/libutil/util.o
nix> clang-7: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
nix> src/libutil/util.cc:1451:30: error: implicit instantiation of undefined template 'std::__1::array<char, 256>'
nix> static std::array<char, 256> base64DecodeChars;
nix>                              ^
nix> /nix/store/w6mp8b25lld3b7s0gvi709dfkw619s36-libcxx-7.1.0-dev/include/c++/v1/__tuple:223:64: note: template is declared here
nix> template <class _Tp, size_t _Size> struct _LIBCPP_TEMPLATE_VIS array;
nix>                                                                ^
nix> 1 error generated.
nix> make: *** [mk/patterns.mk:3: src/libutil/util.o] Error 1
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 136271` at cb9dfc05 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/c24733197d84efd95f1f74ea21401e0a">hydra-unstable</a></li>
    <li><a href="https://gist.github.com/c1c96a29c613d622d52eb374465c9204">nix-update</a></li>
    <li><a href="https://gist.github.com/c2f254e3317d228d29a18846edcde0dc">nixUnstable</a></li>
    <li><a href="https://gist.github.com/8f9ed97c75673e98959d31958bc51311">nixpkgs-review</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -232,15 +232,21 @@ in rec {
...
+    #  - changes in upstream packaging
+    #    git diff OLD..NEW -- $(find . -name \*.nix | grep -v test)
     pname = "nix";
     version = "2.4${suffix}";
-    suffix = "pre20210802_47e96bb";
+    suffix = "pre20210831_${lib.substring 0 7 src.rev}";
 
     src = fetchFromGitHub {
-      owner = "NixOS";
+      owner = "hercules-ci";
```
Comment (happysalada) [inline pkgs/tools/package-management/nix/default.nix:246]: Perhaps just a leftover from testing ?

Comment (roberth) [reply pkgs/tools/package-management/nix/default.nix:246]: It's an unmerged PR. This is actually a draft PR now.

**Comment by happysalada** [comment]
I just noticed it was a draft. On the latest commit, nixUnstable build fine on darwin-x86_64.

**Comment by roberth** [comment]
I changed it after your comment :sweat_smile: 

I'm waiting for upstream to merge a solution. If you feel like taking this over you're welcome to do so.

**Comment by happysalada** [comment]
No worries, I just wanted to help. I'm not in a rush at all on this.
Thanks a lot for doing this!

**Comment by roberth** [comment]
Upstream would rather do their own thing. I'm not spending more time on this.
The commit is tagged downstream so you can use it for pinning. Or you can merge this, or close it and follow up with another fix.

**Comment by happysalada** [comment]
Thank you for having taken a look. Here is the upstream PR for people looking for more information https://github.com/NixOS/nix/pull/5197
The other thing I noticed, is that there seems to be an aarch64-linux CI failure.
It looks like this is going to need some more love. I don't have a lot of time at the moment, but let's leave this open if you don't mind.

**Comment by SuperSandro2000** [comment]
Closing in favor of #136463

**Pull request closed without merging**